pip/download.py

```diff
@@ -117,6 +118,13 @@ def user_agent():
     )
 
 
+def unquote(s):
+    if six.PY2:
+        return urllib_unquote(s.encode("utf-8")).decode("utf-8")
```

Comment:
Thanks for reviewing.
I think because of the `encode("utf-8")` the argument of `urllib_unquote` will be of type str in this case, so  the [_is_unicode](https://github.com/python/cpython/blob/2.7/Lib/urllib.py#L1223) condition will not be met and the bytes will not be decoded as latin1.

This means that unicode characters like the £ used in the unit tests are decoded:
```
>>> urllib.unquote(u'%C2%A3'.encode("utf-8")).decode("utf-8")
u'\xa3'
>>> urllib.unquote('%C2%A3'.encode("utf-8")).decode("utf-8")
u'\xa3'
```
Which wouldn't be decoded otherwise:
```
>>> urllib.unquote(u'%C2%A3')
u'\xc2\xa3'
>>> urllib.unquote('%C2%A3')
'\xc2\xa3'
```